lms/djangoapps/instructor_task/tasks_helper/grades.py

```diff
@@ -700,98 +739,72 @@ def _generate(self, context):
         """
-        return list(header_row.values()) + ['error_msg']
+        return list(self._problem_grades_header().values()) + ['error_msg']
 
-    def _graded_scorable_blocks_to_header(self, course):
-        """
-        Returns an OrderedDict that maps a scorable block's id to its
-        headers in the final report.
-        """
-        scorable_blocks_map = OrderedDict()
-        grading_context = grades_context.grading_context_for_course(course)
-        for assignment_type_name, subsection_infos in six.iteritems(grading_context['all_graded_subsections_by_type']):
-            for subsection_index, subsection_info in enumerate(subsection_infos, start=1):
-                for scorable_block in subsection_info['scored_descendants']:
-                    header_name = (
-                        "{assignment_type} {subsection_index}: "
-                        "{subsection_name} - {scorable_block_name}"
-                    ).format(
-                        scorable_block_name=scorable_block.display_name,
-                        assignment_type=assignment_type_name,
-                        subsection_index=subsection_index,
-                        subsection_name=subsection_info['subsection_block'].display_name,
-                    )
-                    scorable_blocks_map[scorable_block.location] = [header_name + " (Earned)",
-                                                                    header_name + " (Possible)"]
-        return scorable_blocks_map
-
-    def _rows_for_users(self, context, users, header_row, graded_scorable_blocks):
+    def _rows_for_users(self, context, users):
         """
         Returns a list of rows for the given users for this report.
         """
-        if six.text_type(context.course_id) == 'course-v1:MITx+CTL.SC0x+1T2019':
-            context.update_status('ProblemGradeReport Investigation log: Start of row creation for users')
+        self.log_additional_info_for_mit_course(context, 'ProblemGradeReport: Start of row creation for users')
         success_rows, error_rows = [], []
-        for student, course_grade, error in CourseGradeFactory().iter(users, context.course):
-            student_fields = [getattr(student, field_name) for field_name in header_row]
+        for student, course_grade, error in CourseGradeFactory().iter(
+            users,
+            course=context.course,
+            collected_block_structure=context.course_structure,
+            course_key=context.course_id,
+        ):
             context.task_progress.attempted += 1
-            if context.task_progress.attempted % 2 == 0 and \
-                    six.text_type(context.course_id) == 'course-v1:MITx+CTL.SC0x+1T2019':
-                context.update_status('ProblemGradeReport Investigation log: Processing user {}'.format(
-                    context.task_progress.attempted))
+            if context.task_progress.attempted % 2 == 0:
+                self.log_additional_info_for_mit_course(
```

Comment:
Can this be generic? So we have more context if another course grade_report fails?